operators/src/processing/reprojection.rs

```diff
@@ -0,0 +1,399 @@
+        let state = self.state;
+        match self.vector_sources[0].query_processor()? {
+            // TODO: use macro for that
+            TypedVectorQueryProcessor::Data(source) => Ok(TypedVectorQueryProcessor::Data(
+                MapQueryProcessor::new(source, move |query| {
+                    query_rewrite_fn(query, state.0, state.1)
+                })
+                .boxed(),
+            )),
+            TypedVectorQueryProcessor::MultiPoint(source) => {
+                Ok(TypedVectorQueryProcessor::MultiPoint(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+            TypedVectorQueryProcessor::MultiLineString(source) => {
+                Ok(TypedVectorQueryProcessor::MultiLineString(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+            TypedVectorQueryProcessor::MultiPolygon(source) => {
+                Ok(TypedVectorQueryProcessor::MultiPolygon(
+                    ReprojectionProcessor::new(source, self.state.0, self.state.1).boxed(),
+                ))
+            }
+        }
+    }
+}
+
+struct ReprojectionProcessor<Q, G> {
+    source: Q,
+    from: SpatialReference,
+    to: SpatialReference,
+    geom_type: PhantomData<G>,
+}
+
+impl<Q, G> ReprojectionProcessor<Q, G> {
+    pub fn new(source: Q, from: SpatialReference, to: SpatialReference) -> Self {
+        Self {
+            source,
+            from,
+            to,
+            geom_type: PhantomData,
+        }
+    }
+}
+
+/// this method performs the reverse transformation of a query rectangle
+fn query_rewrite_fn(
+    query: QueryRectangle,
+    from: SpatialReference,
+    to: SpatialReference,
+) -> QueryRectangle {
+    let projector = CoordinateProjector::from_known_srs(to, from).unwrap();
+    // TODO: change the resolution...
+    QueryRectangle {
+        bbox: query.bbox.reproject(&projector).unwrap(),
+        ..query
+    }
+}
+
+impl<Q, G> VectorQueryProcessor for ReprojectionProcessor<Q, G>
+where
+    Q: VectorQueryProcessor<VectorType = G>,
+    G: Reproject<CoordinateProjector> + Sync + Send,
+{
+    type VectorType = G::Out;
+
+    fn vector_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> futures::stream::BoxStream<'a, Result<Self::VectorType>> {
+        let rewritten_query = query_rewrite_fn(query, self.from, self.to);
+
+        Box::pin(
+            self.source
+                .vector_query(rewritten_query, ctx)
+                .map_ok(move |collection| {
+                    collection
+                        .reproject(
+                            CoordinateProjector::from_known_srs(self.from, self.to)
+                                .unwrap()
+                                .as_ref(),
+                        )
+                        .unwrap()
```

Comment:
I guess you should use `map` and just propagate the error here.

Reply:
cf493cd7b9b2e28ca9ef5c4a7ac0838552a2591e